app-tasks/rf/src/rf/uploads/geotiff/create_thumbnails.py

```diff
@@ -0,0 +1,137 @@
+"""Creates thumbnails for geotiff imagery
+
+Depends on having geotiff file already downloaded
+"""
+import os
+import subprocess
+
+import boto3
+
+from rf.models import Thumbnail
+
+from .io import get_geotiff_dimensions
+
+def create_thumbnails(tif_path, scene_id, organization_id):
+    """Creates thumbnails based on a geotiff path and scene
+
+    Args:
+        tif_path (str): path of local tiff file
+        scene_id (str): uuid of the scene
+
+    Returns:
+        List[Thumbnail]
+    """
+
+    r_tif_path = 'resampled_{}'.format(tif_path)
+    rp_tif_path = 'reprojected_{}'.format(tif_path)
+
+    dim = get_geotiff_dimensions(tif_path)
+    max_dim = float(max(get_geotiff_dimensions(tif_path)))
+
+    size_large = 1024
+    size_small = 256
+
+    scale_large = size_large / max_dim
+    scale_small = size_small / max_dim
+
+    dim_large = (dim[0] * scale_large, dim[1] * scale_large)
+    dim_small = (dim[0] * scale_small, dim[1] * scale_small)
+
+    # Create paths for each size thumb
+    path_large = '{}-LARGE.png'.format(scene_id)
+    path_small = '{}-SMALL.png'.format(scene_id)
+
+    # Create urls for each size Thumbnail
+    url_large = '/thumbnails/{}'.format(path_large)
+    url_small = '/thumbnails/{}'.format(path_small)
+
+    try_to_remove_files([r_tif_path, rp_tif_path, path_large, path_small])
+
+    if os.path.isfile(tif_path):
+        try:
+            # Resample tif
+            subprocess.check_call([
+                'gdalwarp', tif_path, r_tif_path,
+                '-ts', str(dim_large[0]), str(dim_large[1]),
+                '-q'
+            ])
+
+            # Reproject tif
+            subprocess.check_call([
+                'gdalwarp', r_tif_path, rp_tif_path,
+                '-t_srs', 'EPSG:3857',
+                '-q'
+            ])
+
+            # Create a temporary env object
+            mod_env = os.environ.copy()
+            # Add variable to avoid sidecar files
+            mod_env['GDAL_PAM_ENABLED'] = 'NO'
+
+            # Convert tif to pngs (large)
+            subprocess.check_call([
+                'gdal_translate', rp_tif_path, path_large,
+                '-b', '1',
+                '-b', '2',
+                '-b', '3',
+                '-outsize', str(dim_large[0]), str(dim_large[1]),
+                '-of', 'PNG',
+                '-q'
+            ], env=mod_env)
+
+            # Convert tif to pngs (small)
+            subprocess.check_call([
+                'gdal_translate', rp_tif_path, path_small,
+                '-b', '1',
+                '-b', '2',
+                '-b', '3',
+                '-outsize', str(dim_small[0]), str(dim_small[1]),
+                '-of', 'PNG',
+                '-q',
+            ], env=mod_env)
+
+        except:
```

Comment:
We only want to perform the cleanup if one of the subprocess calls fails and not when they all succeed, so I think this solution is OK.